compiler/rustc.vim

```diff
@@ -21,7 +21,11 @@ endif
 if get(g:, 'rustc_makeprg_no_percent', 0)
     CompilerSet makeprg=rustc
 else
-    CompilerSet makeprg=rustc\ \%
+    if has('patch-7.4.191')
+      CompilerSet makeprg=rustc\ \%:S
+    else
+      exe 'CompilerSet makeprg=rustc ' . shellescape(expand('%'))
```

Comment:
I meant just doing without this, `CompilerSet makeprg=rustc\ \%`. You can’t trust `%` to remain the same over time (e.g. start with `:new | setf rust` and only give it a filename later).

Reply:
Exactly, 

> In case the patch is not included, then the file path is expanded and shell escaped. Thus there is no perfect solution: if the file is renamed, &makeprg uses a non-existent file path. If the file path is not escaped, then possibly as well.

There is no perfect solution.

Reply:
Ah yes, I didn’t read very thoroughly, did I? But yeah, I’d prefer it to just skip the escaping. Simpler (which is important for legacy code that isn’t the normal code path), typically still right.

Reply:
Okay. Let's hope for the best! Coders rarely use file paths with anything but alphanumeric letters. Different for markup languages.